platform/ui/src/components/SidePanel/SidePanel.tsx

```diff
@@ -127,6 +130,23 @@ const SidePanel = ({
     [updatePanelOpen]
   );
 
+  useEffect(() => {
+    const { unsubscribe } = panelService.subscribe(
+      panelService.EVENTS.PANELS_CHANGED,
+      panelChangedEvent => {
+        if (panelChangedEvent.position !== side) {
+          return;
+        }
+
+        setTabs(panelService.getPanels(side));
+      }
+    );
+
+    return () => {
+      unsubscribe();
+    };
+  }, [panelService, side]);
+
```

Comment:
it is getting a lot coupled with panelService, an we refactor this to take out the logic for ui separate, and we call the component that has service something like `SidePanelWithService`

Reply:
I added a SidePanelWithService as requested but I'm not sure where it would be (platform? extension? another folder?)